src/mds/MDCache.cc

```diff
@@ -9248,6 +9248,32 @@ void MDCache::request_forward(MDRequestRef& mdr, mds_rank_t who, int port)
     dout(7) << "request_forward " << *mdr << " to mds." << who << " req "
             << *mdr->client_request << dendl;
     mds->forward_message_mds(mdr->release_client_request(), who);
+    if (mdr->is_batch_head) {
+      BatchOpContext* fin = NULL;
+      int mask = mdr->client_request->head.args.getattr.mask;
+
+      if (mdr->client_request->get_op() == CEPH_MDS_OP_GETATTR) {
+	CInode* in = mdr->in[0];
+	if (in) {
+	  if (in->batch_ops.count(mask)) {
+	    fin = in->batch_ops[mask];
+	    in->batch_ops.erase(mask);
+	  }
+	}
+      } else {
+	CDentry* dn = mdr->dn[0].back();
+	if (dn) {
+	  if (dn->batch_ops.count(mask)) {
+	    fin = dn->batch_ops[mask];
+	    dn->batch_ops.erase(mask);
```

Comment:
code like this does map search 3 times. following code only do search one time

auto it = dn->batch_ops.find(mask);
if (it != dn->batch_ops.end()) {
find = it->second;
dn->batch_ops.erase(it);
}

Reply:
Oh, yeah, you are right:-) Correcting it:-)